### PR TITLE
Fixed print() statements for CLUE display in debug mode

### PR DESCRIPTION
**Pull request by andreamah**
# Description:

Previously, print statements did not show on the CLUE's display while in debug mode. I added a wrapper for `print()` in debug_user_code.py so that prints would also be sent to the clue display.

I also added starting and ending print statements to each run to match non-debug mode.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

# Testing:

Prints should appear both on the clue screen and in vscode's terminal.
```
from adafruit_clue import clue
import datetime
import time

clue_data = clue.simple_text_display(title="Data Screen!", title_scale=1, text_scale=3)

val = 100
for i in range(12):
    val += 1
    clue_data.show_terminal()
    time.sleep(0.5)
    print(val)
    print(f"time {datetime.datetime.now(datetime.timezone.utc).strftime('%Y-%m-%d %H:%M:%S.%f')[:-3]}")


for i in range(10):
    val += 1
    clue_data[3].text = "Count: {}".format(val)
    clue_data.show()
    print("shouldn't show until the end")
    time.sleep(0.5)
    
```
# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] My code has been formatted with `npm run format` and passes the checks in `npm run check`
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by sagarmanchanda — APPROVED** [review]
lgtm :)

File: src/debug_user_code.py [hunk 1]
```diff
@@ -56,12 +62,26 @@
...
+    global curr_terminal
+
+    def wrapped_func(*args, **kwargs):
+        curr_terminal.add_str_to_terminal("".join(str(e) for e in args))
+        return func(*args, **kwargs)
+
+    return wrapped_func
+
+
+print = print_decorator(print)
```
Comment (vandyliu) [inline src/debug_user_code.py:76]: do we need to restore print in the end?

Comment (andreamah) [reply src/debug_user_code.py:76]: That's us using the print_decorator function to override print with our custom version of print.

Comment (vandyliu) [reply src/debug_user_code.py:76]: should we restore it though after debug_user_code is ran?
so like
original_print = print
print = print_decorator(print)

...
print = original_print

Comment (andreamah) [reply src/debug_user_code.py:76]: In what case would this needed? When the program is done, all globals are reset.

Comment (vandyliu) [reply src/debug_user_code.py:76]: I guess so. Thanks Andrea

**Review by vandyliu — APPROVED** [review]
Lgtm

**Pull request merged**